From 2f4918f22ea4b148bb9208e020ce9214c1cc3147 Mon Sep 17 00:00:00 2001
From: nhannguyen <nhan.nguyen.yb@renesas.com>
Date: Wed, 15 Nov 2017 18:05:51 +0700
Subject: [PATCH 109/504] ethavb: Fix changed PHY address on Board iWave G1E.

This is a workaround to fix PHY address Board iWave G1E 's PHY address is changed of 1 or 3.
This uses a loop to check the PHY address

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/of/of_mdio.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/of/of_mdio.c b/drivers/of/of_mdio.c
index 2b1ccb8..f1f1b89 100644
--- a/drivers/of/of_mdio.c
+++ b/drivers/of/of_mdio.c
@@ -54,8 +54,15 @@ static int of_mdiobus_register_phy(struct mii_bus *mdio, struct device_node *chi
 
 	if (!is_c45 && !of_get_phy_id(child, &phy_id))
 		phy = phy_device_create(mdio, addr, phy_id, 0, NULL);
-	else
-		phy = get_phy_device(mdio, addr, is_c45);
+	else {
+		for (addr = 1; addr < PHY_MAX_ADDR ; addr++) {
+			phy = get_phy_device(mdio, addr, is_c45);
+			if (!phy || IS_ERR(phy))
+				continue;
+			else
+				break;
+		}
+	}
 	if (!phy || IS_ERR(phy))
 		return 1;
 
-- 
2.7.4

